scripts/ci_checks/linux-x86_64/ubuntu-24.04.sh

```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env bash
+
+set -euxo pipefail
+
+for comp in gcc-12 gcc-13 clang-16 clang-17
```

Comment:
Let's remove gcc-12 (already covered by other image) and replace clang 16 with 15, because clang 15 is available in more ubuntu versions than 16 and so is more important to cover I think.